opm/input/eclipse/Schedule/Schedule.cpp

```diff
@@ -580,30 +581,47 @@ class ScheduleLogger
 
 namespace
 {
-/// \brief Check whether each MS well has COMPSEGS entry andissue error if not.
-/// \param welsegs All wells with a WELSEGS entry together with the location.
-/// \param compsegs All wells with a COMPSEGS entry
-void check_compsegs_consistency(const Opm::WelSegsSet& welsegs,
-                                const std::set<std::string>& compsegs,
-                                const std::vector<::Opm::Well>& wells)
-{
-    const auto difference = welsegs.difference(compsegs, wells);
 
-    if (!difference.empty()) {
+void report_welsegs_error(const std::vector<Opm::WelSegsSet::Entry>& segments, std::string format_str) 
+{
+    if (!segments.empty()) {
         std::string well_str = "well";
-        if (difference.size() > 1) {
+        if (segments.size() > 1) {
             well_str.append("s");
         }
         well_str.append(":");
 
-        for(const auto& [name, location] : difference) {
+        for(const auto& [name, location] : segments) {
             well_str.append(fmt::format("\n   {} in {} at line {}",
                                         name, location.filename, location.lineno));
         }
-        auto msg = fmt::format("Missing COMPSEGS keyword for the following multisegment {}.", well_str);
-        throw Opm::OpmInputError(msg, std::get<1>(difference[0]));
+        auto msg = fmt::format(format_str, well_str);
+        throw Opm::OpmInputError(msg, std::get<1>(segments[0]));
     }
 }
+
+/// \brief Check whether each MS well has COMPSEGS entry and issue error if not.
+/// \param welsegs All wells with a WELSEGS entry together with the location.
+/// \param compsegs All wells with a COMPSEGS entry
+/// \param comptraj All wells with a COMPTRAJ entry
+void check_compsegs_and_comptraj_consistency(const Opm::WelSegsSet& welsegs,
+                                             const std::set<std::string>& compsegs,
+                                             const std::set<std::string>& comptraj,
+                                             const std::vector<::Opm::Well>& wells)
+{
+    std::set<std::string> comp = compsegs;
+    comp.insert(comptraj.begin(), comptraj.end());
```

Comment:
you can use std::set_union for this, which is clearer. 

```c++
std::set<std::string> comp;
std::set_union(compsegs.begin(), compsegs.end(),
               comptraj.begin(), comptraj.end(),
               std::inserter(comp, comp.begin()));
```

`comp` can use a more descriptive name.